File: void-lib/void-orm-lib/void-standard-orm-lib/void-base-orm/src/main/java/cn/toutatis/xvoid/orm/base/authentication/service/VoidAuthService.java
package cn.toutatis.xvoid.orm.base.authentication.service;

import cn.toutatis.xvoid.common.exception.AuthenticationException;
import cn.toutatis.xvoid.orm.base.authentication.entity.RequestAuthEntity;

/**
 * Void auth service
 * 认证服务接口
 * @constructor Create empty Void auth service
 */
public interface VoidAuthService {

    /**
     * Pre check
     * 用户名预检查
     * @param requestAuthEntity 用户实体
     * @return 账户是否存在
     */
    boolean preCheckAccount(RequestAuthEntity requestAuthEntity);

    /**
     * Throw failed
     * 抛出异常
     * @param message 异常消息
     */
    default void throwFailed(String message){
        throw new AuthenticationException(message);
    };
}

File: void-spring/src/main/kotlin/cn/toutatis/xvoid/spring/core/security/access/auth/VoidAuthService.kt
package cn.toutatis.xvoid.spring.core.security.access.auth

import cn.toutatis.xvoid.common.exception.AuthenticationException
import cn.toutatis.xvoid.spring.core.security.access.AuthType
import com.alibaba.fastjson.JSONObject
import kotlin.jvm.Throws

/**
 * Void auth service
 * 认证服务接口
 * @constructor Create empty Void auth service
 */
interface VoidAuthService {

    /**
     * Pre check
     * 用户名预检查
     * @param account 用户名
     * @param authType 账户类型
     * @return 账户是否存在
     */
    fun preCheckAccount(identity:JSONObject):Boolean

    /**
     * Throw failed
     * 抛出异常
     * @param message 异常消息
     */
    @Throws(AuthenticationException::class)
    fun throwFailed(message:String): Unit = throw AuthenticationException(message)

}